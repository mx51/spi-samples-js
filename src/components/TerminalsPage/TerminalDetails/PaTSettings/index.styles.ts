import { createStyles, makeStyles, Theme } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    sectionSpacing: {
      marginTop: theme.spacing(3),
      paddingTop: theme.spacing(3),
    },
    switchBox: {
      padding: theme.spacing(1.5, 1),
      marginBottom: theme.spacing(1.5),
      '&>.MuiTypography-root': {
        display: 'flex',
        alignItems: 'center',
      },
    },
    switchBoxTitle: {
      fontSize: '1.125rem',
      fontWeight: 500,
      wordBreak: 'break-word',
    },
    switchBoxSubtitle: {
      fontSize: '0.875rem',
      fontWeight: 400,
      color: theme.palette.text.secondary,
      wordBreak: 'break-word',
    },
  })
);

export default useStyles;
