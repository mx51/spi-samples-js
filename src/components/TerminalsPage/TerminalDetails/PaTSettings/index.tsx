import React from 'react';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import Switch from '@material-ui/core/Switch';
import Typography from '@material-ui/core/Typography';
import { payAtTableFields, payAtTableOptions } from '../../../../definitions/constants/terminalConfigs';
import CustomTextField from '../../../CustomTextField';
import useStyles from './index.styles';

export default function PaTSettings(): React.ReactElement {
  const classes = useStyles();

  return (
    <>
      <Grid container className={classes.sectionSpacing}>
        <Grid item md={4} xs={12}>
          <Typography>Settings</Typography>
        </Grid>
        <Grid item md={8} xs={12}>
          {payAtTableOptions.map(({ name, title, subtitle }) => (
            <Box alignItems="center" className={classes.switchBox} display="flex" key={name}>
              <Typography align="right">
                <Switch name={name} inputProps={{ 'aria-label': title }} />
              </Typography>
              <Box display="flex" flexDirection="column" marginLeft={2}>
                <Typography variant="h5">{title}</Typography>
                <Typography variant="inherit">{subtitle}</Typography>
              </Box>
            </Box>
          ))}
        </Grid>
      </Grid>
      <Grid container className={classes.sectionSpacing}>
        <Grid item md={4} xs={12}>
          <Typography>Customisation</Typography>
        </Grid>
        <Grid item md={8} xs={12}>
          {payAtTableFields.map(({ helperText, id, label }) => (
            <CustomTextField
              key={id}
              fullWidth
              helperText={helperText}
              dataTestId={id}
              label={label}
              margin="dense"
              value=""
              variant="outlined"
            />
          ))}
        </Grid>
      </Grid>
    </>
  );
}
