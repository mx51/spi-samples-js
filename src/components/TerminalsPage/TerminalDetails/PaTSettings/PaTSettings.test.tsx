import React from 'react';
import { cleanup } from '@testing-library/react';
import PaTSettings from '.';
import mockWithRedux from '../../../../utils/tests/common';

describe('Test <PaTSettings />', () => {
  afterEach(cleanup);

  test('should match PaTSettings snapshot test', () => {
    // Arrange
    const container = mockWithRedux(<PaTSettings />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
