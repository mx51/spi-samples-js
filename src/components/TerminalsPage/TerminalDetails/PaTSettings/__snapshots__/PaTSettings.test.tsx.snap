// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PaTSettings /> should match PaTSettings snapshot test 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-sectionSpacing-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Settings
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    >
      <div
        class="MuiBox-root MuiBox-root-5 makeStyles-switchBox-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          <span
            class="MuiSwitch-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-label="Pay at Table"
                  class="PrivateSwitchBase-input-9 MuiSwitch-input"
                  name="payAtTable"
                  type="checkbox"
                  value=""
                />
                <span
                  class="MuiSwitch-thumb"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track"
            />
          </span>
        </p>
        <div
          class="MuiBox-root MuiBox-root-10"
        >
          <h5
            class="MuiTypography-root makeStyles-switchBoxTitle-3 MuiTypography-h5"
          >
            Pay at Table
          </h5>
          <span
            class="MuiTypography-root makeStyles-switchBoxSubtitle-4"
          >
            Enable Pay at Table for this terminal
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-11 makeStyles-switchBox-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          <span
            class="MuiSwitch-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-label="Split bill evenly"
                  class="PrivateSwitchBase-input-9 MuiSwitch-input"
                  name="splitBillEvenly"
                  type="checkbox"
                  value=""
                />
                <span
                  class="MuiSwitch-thumb"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track"
            />
          </span>
        </p>
        <div
          class="MuiBox-root MuiBox-root-12"
        >
          <h5
            class="MuiTypography-root makeStyles-switchBoxTitle-3 MuiTypography-h5"
          >
            Split bill evenly
          </h5>
          <span
            class="MuiTypography-root makeStyles-switchBoxSubtitle-4"
          >
            Allow customers to split the bill equally
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-13 makeStyles-switchBox-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          <span
            class="MuiSwitch-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-label="Tipping"
                  class="PrivateSwitchBase-input-9 MuiSwitch-input"
                  name="tipping"
                  type="checkbox"
                  value=""
                />
                <span
                  class="MuiSwitch-thumb"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track"
            />
          </span>
        </p>
        <div
          class="MuiBox-root MuiBox-root-14"
        >
          <h5
            class="MuiTypography-root makeStyles-switchBoxTitle-3 MuiTypography-h5"
          >
            Tipping
          </h5>
          <span
            class="MuiTypography-root makeStyles-switchBoxSubtitle-4"
          >
            Allow customers to tip during a Pay at Table transaction
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-15 makeStyles-switchBox-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          <span
            class="MuiSwitch-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-label="Accept cash"
                  class="PrivateSwitchBase-input-9 MuiSwitch-input"
                  name="acceptCash"
                  type="checkbox"
                  value=""
                />
                <span
                  class="MuiSwitch-thumb"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track"
            />
          </span>
        </p>
        <div
          class="MuiBox-root MuiBox-root-16"
        >
          <h5
            class="MuiTypography-root makeStyles-switchBoxTitle-3 MuiTypography-h5"
          >
            Accept cash
          </h5>
          <span
            class="MuiTypography-root makeStyles-switchBoxSubtitle-4"
          >
            Allow customers to pay by cash during a Pay at Table transaction
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-17 makeStyles-switchBox-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          <span
            class="MuiSwitch-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-label="Retrieve all tables"
                  class="PrivateSwitchBase-input-9 MuiSwitch-input"
                  name="retrieveAllTables"
                  type="checkbox"
                  value=""
                />
                <span
                  class="MuiSwitch-thumb"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track"
            />
          </span>
        </p>
        <div
          class="MuiBox-root MuiBox-root-18"
        >
          <h5
            class="MuiTypography-root makeStyles-switchBoxTitle-3 MuiTypography-h5"
          >
            Retrieve all tables
          </h5>
          <span
            class="MuiTypography-root makeStyles-switchBoxSubtitle-4"
          >
            Request a list of all open tables and sales from the POS
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-19 makeStyles-switchBox-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          <span
            class="MuiSwitch-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-label="Require Operator ID"
                  class="PrivateSwitchBase-input-9 MuiSwitch-input"
                  name="requireOperatorID"
                  type="checkbox"
                  value=""
                />
                <span
                  class="MuiSwitch-thumb"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track"
            />
          </span>
        </p>
        <div
          class="MuiBox-root MuiBox-root-20"
        >
          <h5
            class="MuiTypography-root makeStyles-switchBoxTitle-3 MuiTypography-h5"
          >
            Require Operator ID
          </h5>
          <span
            class="MuiTypography-root makeStyles-switchBoxSubtitle-4"
          >
            Prompt for Operator ID in Pay at Table mode
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-21 makeStyles-switchBox-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          <span
            class="MuiSwitch-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-label="Transaction summary report"
                  class="PrivateSwitchBase-input-9 MuiSwitch-input"
                  name="transactionSummaryReport"
                  type="checkbox"
                  value=""
                />
                <span
                  class="MuiSwitch-thumb"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track"
            />
          </span>
        </p>
        <div
          class="MuiBox-root MuiBox-root-22"
        >
          <h5
            class="MuiTypography-root makeStyles-switchBoxTitle-3 MuiTypography-h5"
          >
            Transaction summary report
          </h5>
          <span
            class="MuiTypography-root makeStyles-switchBoxSubtitle-4"
          >
            Print a summary at the end of a Pay at Table transaction
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-sectionSpacing-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Customisation
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    >
      <div
        class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-23 MuiFormControl-marginDense MuiFormControl-fullWidth"
        data-test-id="payAtTableLabelField"
        theme="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink="false"
        >
          Pay at Table label
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                Pay at Table label
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-marginDense"
        >
          Customise the ‘Pay at Table’ button displayed on the terminal
        </p>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-23 MuiFormControl-marginDense MuiFormControl-fullWidth"
        data-test-id="operatorIDField"
        theme="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink="false"
        >
          Operator ID label
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                Operator ID label
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-marginDense"
        >
          Customise the ‘Operator ID’ field label displayed on the terminal
        </p>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-23 MuiFormControl-marginDense MuiFormControl-fullWidth"
        data-test-id="tableLabelField"
        theme="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink="false"
        >
          Table label
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                Table label
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-marginDense"
        >
          Customise the ‘Table’ field label displayed on the terminal
        </p>
      </div>
    </div>
  </div>
</div>
`;
